pkg/sdn/plugin/node.go

```diff
@@ -120,18 +128,74 @@ func (node *OsdnNode) Start() error {
 		}
 		for _, p := range pods {
 			containerID := getPodContainerID(&p)
-			err = node.UpdatePod(p.Namespace, p.Name, kubeletTypes.ContainerID{ID: containerID})
 			if err != nil {
+				err = node.UpdatePod(p.Namespace, p.Name, kubeletTypes.ContainerID{ID: containerID})
 				log.Warningf("Could not update pod %q (%s): %s", p.Name, containerID, err)
 			}
 		}
 	}
 
+	if err := node.cniServer.Start(); err != nil {
+		return err
+	}
+	go utilwait.Forever(node.watchCni, 0)
+
 	node.markPodNetworkReady()
 
 	return nil
 }
 
+func (node *OsdnNode) watchCni() {
+	for {
+		select {
+		case request := <-node.podRequestChan:
```

Comment:
It looks like this loop can block if any CNI container fails.  You'd need to have independent pod queues and handle them separately, and deal with CNI clients that crash and can't receive their updates.  It would be better to preserve all of the updates for each pod in a local cache and identify which to serve to the client (the latest).